src/main/java/com/sap/oss/phosphor/fosstars/data/owasp/experimental/VulnerabilitiesFromOwaspDependencyCheck.java

```diff
@@ -0,0 +1,404 @@
+    return this;
+  }
+
+  /**
+   * Returns the supported feature loaded by this data provider.
+   */
+  @Override
+  public Set<Feature<?>> supportedFeatures() {
+    return setOf(VULNERABILITIES);
+  }
+
+  /**
+   * Scan the input jar file and analyze the extracted {@link Dependency}.
+   * 
+   * @param file The jar.
+   */
+  private void analyze(File file) {
+    try {
+      engine.scan(file);
+      engine.analyzeDependencies();
+    } catch (ExceptionCollection e) {
+      exceptionCollection.addException(e);
+    }
+  }
+
+  /**
+   * Process the report generated from analysis of the jar file.
+   * 
+   * @param fileName The name of report.
+   * @return An optional of {@link OwaspDependencyCheckEntry}.
+   * @throws IOException If something went wrong.
+   */
+  private Optional<OwaspDependencyCheckEntry> process(String fileName) throws IOException {
+    Optional<Path> reportPath = createDirectory(REPORT_DIR);
+
+    if (reportPath.isPresent()) {
+      File file = reportPath.get().resolve(String.format("%s.json", fileName)).toFile();
+
+      try {
+        engine.writeReports(fileName, file, REPORT_OUTPUT_FORMAT, exceptionCollection);
+        ObjectMapper mapper = new ObjectMapper();
+        return Optional.ofNullable(mapper.readValue(file, OwaspDependencyCheckEntry.class));
+      } catch (ReportException e) {
+        throw new IOException("Oh no! the report writing failed.", e);
+      }
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * Scan the {@link MavenArtifact}.
+   * 
+   * @param artifact The {@link MavenArtifact}.
+   * @return An optional of {@link OwaspDependencyCheckEntry}.
+   * @throws IOException If something went wrong.
+   */
+  private Optional<OwaspDependencyCheckEntry> scan(MavenArtifact artifact) throws IOException {
+    Optional<Path> filePath = fetch(artifact);
+
+    if (filePath.isPresent()) {
+      try {
+        analyze(filePath.get().toFile());
+        return process(filePath.get().toFile().getName());
+      } finally {
+        if (engine != null) {
+          engine.close();
+        }
+      }
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * Converts an {@link OwaspDependencyCheckVuln} to a {@link Vulnerability}.
+   *
+   * @param owaspDependencyCheckVuln The {@link OwaspDependencyCheckVuln} to be converted.
+   * @return An instance of {@link Vulnerability}.
+   * @throws IOException If something went wrong.
+   */
+  private static Vulnerability from(OwaspDependencyCheckVuln owaspDependencyCheckVuln)
+      throws IOException {
+    Objects.requireNonNull(owaspDependencyCheckVuln,
+        "Oh no! vulnerability from OWASP entry is null");
+
+    return Builder.newVulnerability(owaspDependencyCheckVuln.getName())
+        .description(owaspDependencyCheckVuln.getDescription())
+        .set(cvssFrom(owaspDependencyCheckVuln))
+        .references(referencesFrom(owaspDependencyCheckVuln))
+        .set(Resolution.PATCHED)
```

Comment:
Unfortunately, not all vulnerabilities are patched.

Reply:
I feel there is a way to improve this. I have observed that we have been following this assumption for some time now. Let's take it up here #574

Reply:
A usual use-case of OWASP Dependency Check is to find known vulnerabilities for an artifact. Here "known" means unpatched. If the data provider ask Dependency Check to show only known vulnerabilities for an artifact, then it can just set resolution to unpatched.

Reply:
Now I read the OWASP Dependency-Check tool docs. It does say that it reports vulnerabilities affecting the version. It makes sense if we give an `artifact:version` as the input.

But I think #574 can still be used to improve the `Vulnerability` class while collecting vulnerabilities list from `NVD`.

Reply:
I am still not sure what exactly you'd like to do in that new task. If you'd like to search for unpatched vulnerabilities in a project, I've already attempted that. The attempt failed. You can give it another try, no problem here.